app/models/category.rb

```diff
@@ -0,0 +1,33 @@
+class Category < ApplicationRecord
+  validates :title, presence: true
+  validates :title, uniqueness: { scope: :parent_category }, allow_blank: true, if: :title?
+  before_save :parent_category_should_exist?
+  before_save :parent_category_should_not_be_sub_category
```

Comment:
Always mention before the block

# Validations
......

# Callbacks
....


# Associations